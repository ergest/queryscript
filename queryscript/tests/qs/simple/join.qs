import * from inference;

let foo = select * from (select * from users join events on users.id = events.user_id);

SELECT user_id FROM foo;
