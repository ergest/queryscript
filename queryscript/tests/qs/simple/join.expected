{
    "compile_errors": [],
    "decls": {
        "let events": External<[{
        	description Utf8,
        	ts Utf8,
        	user_id Int64,
        }]>,
        "let foo": [{
        	description Utf8,
        	ts Utf8,
        	user_id Int64,
        	active Boolean,
        	id Int64,
        	name Utf8,
        	org_id Int64,
        }],
        "let users": External<[{
        	active Boolean,
        	id Int64,
        	name Utf8,
        	org_id Int64,
        }]>,
    },
    "queries": [
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "user_id",
                                type_: Atom(
                                    Int64,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| user_id |\n|---------|\n| 1       |\n| 2       |\n| 1       |\n| 2       |",
            },
        ),
    ],
}